data/registry_model/registry_oci_model.py

```
@@ -430,6 +439,151 @@ def create_manifest_and_retarget_tag(
+                return result
+
+        logger.warning(
+            "Failed to retarget signature tag '%s' after %d attempts",
+            tag_name,
+            SIGNATURE_RETARGET_MAX_RETRIES,
+        )
+
+        if raise_on_error:
+            raise RetargetTagException(
+                f"Failed to retarget signature tag after {SIGNATURE_RETARGET_MAX_RETRIES} attempts"
+            )
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)